lib/uplink/ext/apikey_test.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/lib/uplink"
+)
+
+func TestParseAPIKey(t *testing.T) {
+	var cErr Cchar
+	apikeyString := "testapikey123"
+	cAPIKeyRef := ParseAPIKey(stringToCCharPtr(apikeyString), &cErr)
+	require.Empty(t, cCharToGoString(cErr))
+
+	apikey, ok := structRefMap.Get(token(cAPIKeyRef)).(uplink.APIKey)
```

Comment:
Test the external behavior rather than internals, testing internals are usually fragile.